.github/workflows/nightly.yml

```diff
@@ -29,16 +30,89 @@ jobs:
         with:
           repository: temporalio/docker-compose
           path: docker-compose
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
 
       - name: Start Temporal Server
         run: docker-compose up -d
         working-directory: docker-compose
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
 
       - name: Wait for Temporal Server
         run: npm run wait-namespace
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
       # end:docker-compose
 
-      - run: npm ci-stress
+      - run: npm run ci-stress
+        env:
+          TEMPORAL_TESTING_SERVER_URL: 'localhost:7233'
+
+  long-run:
+    runs-on: [self-hosted, linux]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+      - uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node }}
+      # Don't build during install phase since we're going to explicitly build
+      - run: npm ci --ignore-scripts
+      - run: npm run build
+        env:
+          BUILD_CORE_RELEASE: true
+
+      # docker-compose
+      - name: Checkout docker-compose repo
+        uses: actions/checkout@v2
+        with:
+          repository: temporalio/docker-compose
+          path: docker-compose
+
+      - name: Start Temporal Server
+        run: docker-compose up -d
+        working-directory: docker-compose
+
+      - name: Wait for Temporal Server
+        run: npm run wait-namespace
+      # end:docker-compose
+
+      - run: npm run ci-nightly-longrun
+        env:
+          TEMPORAL_TESTING_SERVER_URL: 'localhost:7233'
+
+  sampler:
```

Comment:
Purpose? Maybe link @ top of yml files to a cross-sdk testing doc, like `sdk-features/docs/testing.md`